gdrive_sync/api.py

```diff
@@ -485,3 +490,58 @@ def update_sync_status(website: Website, sync_datetime: datetime):
     website.sync_status = new_status
     website.sync_errors = (website.sync_errors or []) + errors
     website.save()
+
+
+@transaction.atomic
+def rename_file(obj_text_id, obj_new_filename):
+    """Rename the file on S3 associated with the WebsiteContent object to a new filename."""
+    obj = WebsiteContent.objects.get(text_id=obj_text_id)
+    site = obj.website
+    df = DriveFile.objects.get(resource=obj)
+    s3 = get_boto3_resource("s3")
+    # slugify just the provided name and then make the extensions lowercase
+    filepath = Path(obj_new_filename)
+    basename = obj_new_filename.rstrip("".join(filepath.suffixes))
+    new_filename = slugify(basename)
+    if filepath.suffixes:
+        new_filename += "".join(filepath.suffixes).lower()
+    df_path = df.s3_key.split("/")
+    df_path[-1] = new_filename
+    new_key = "/".join(df_path)
+    # check if an object with the new filename already exists in this course
+    existing_obj = WebsiteContent.objects.filter(Q(website=site) & Q(file=new_key))
+    if existing_obj:
+        old_obj = existing_obj.first()
+        if old_obj == obj:
+            log.error("New filename is the same as the existing filename.")
+            return
+        dependencies = WebsiteContent.objects.filter(
+            Q(website=site)
+            & Q(type=CONTENT_TYPE_PAGE)
+            & Q(markdown__icontains=old_obj.text_id),
+        )
+        if dependencies:
+            log.error(
```

Comment:
`raise Exception("Not renaming file .... etc`

Reply:
Updated in d4623bd275e9e79b46059736e36cdddb7e916fe4.